### ISSUE TITLE
[Bug]: 'React' must be in scope when using JSXeslintreact/react-in-jsx-scope

### ISSUE THREAD

**Issue by vaynevayne** [jsx-eslint/eslint-plugin-react#3770]
### Is there an existing issue for this?

- [X] I have searched the existing issues and my issue is unique
- [X] My issue appears in the command-line and not only in the text editor

### Description Overview

The eslint rule I just created, react18, actually prompts you to introduce react at the top. This should be the default option, right?
Is this the rule intentional and I need to turn it off manually, or is there a problem with my configuration?

<img width="583" alt="image" src="https://github.com/jsx-eslint/eslint-plugin-react/assets/124666577/7d7c009d-003c-4bea-a745-d959cadc2823">

```jsx
import { fixupConfigRules } from "@eslint/compat";
import pluginJs from '@eslint/js';
import pluginReactConfig from 'eslint-plugin-react/configs/recommended.js';
import globals from 'globals';
import tseslint from 'typescript-eslint';

export default [
  { languageOptions: { globals: globals.browser } },
  pluginJs.configs.recommended,
  ...tseslint.configs.recommended,
  {
    files: ['**/*.jsx'],
    languageOptions: { parserOptions: { ecmaFeatures: { jsx: true } } },
  },
  ...fixupConfigRules(pluginReactConfig),
  {
    rules: {
      '@typescript-eslint/no-explicit-any': 'off',
    },
  },
]
```

### Expected Behavior

no error 

### eslint-plugin-react version

7.34.3

### eslint version

^9.5.0

### node version

v20.14.0

**Comment by ljharb** [comment]
This is by design - you need to extend the `jsx-runtime` config, as detailed in the readme [here](https://github.com/jsx-eslint/eslint-plugin-react?tab=readme-ov-file#configuration-legacy-eslintrc-)

**Comment by GabenGar** [comment]
Not sure by which design, since [getting started section](https://eslint.org/docs/latest/use/getting-started) of ESLint results in this config, so they clearly didn't get the memo:
```javascript
import globals from "globals";
import pluginJs from "@eslint/js";
import tseslint from "typescript-eslint";
import pluginReact from "eslint-plugin-react";

export default [
  {files: ["./src/**/*.{js,mjs,cjs,ts,jsx,tsx}"]},
  {languageOptions: { globals: globals.browser }},
  pluginJs.configs.recommended,
  ...tseslint.configs.recommended,
  pluginReact.configs.flat.recommended,
];
```
And this instantly throws `'React' must be in scope when using JSX  react/react-in-jsx-scope` and `Warning: React version not specified in eslint-plugin-react settings. See https://github.com/jsx-eslint/eslint-plugin-react#configuration` when applied to code which started off [React Remix getting started section](https://remix.run/docs/en/main/start/quickstart) (which might or might not be related).
I am glad to know setting up tooling for react is as arcane as when I only started learning web development.

**Comment by ljharb** [comment]
You’re reading the “flat config” section - read the entire readme from top down, without skipping anything, and you’ll see the section i also linked to directly in the previous comment.

**Comment by GabenGar** [comment]
The part you link to specifically mentions legacy config which I instantly skipped because I use hot new ESLint 9 flat config. There is also no mention of "extending JSX runtime" in the flat config section.
It mentions "new" "JSX transform from React 17", but it's not obvious what is this to a newcomer which arrived there from the error caused by template config. Not to mention it's hard to ascertain how "new" it is, considering react 17 was released [4 years ago](https://legacy.reactjs.org/blog/2020/10/20/react-v17.html), which is pretty much prehistoric times in javascript land, and in react land might as well be a deprecated legacy bad practice by now. It even predates COVID-19, so it was quite literally released in a different world in a general historic context.
All I am saying a niche plugin for a lib shouldn't require reading a pretty long documentation with a lot of noise just to work at all on a template config.
